.codespellrc

```diff
@@ -0,0 +1,5 @@
+[codespell]
+skip = .git,*.pdf,*.svg,*.patch,*.py,*.diff,*.pub,foamEtcFile,patch.*
```

Comment:
Is there a way to skip the intentional non-utf-8 test files like `lib/spack/spack/test/data/filter_file/x86_cpuid_info.c`? It flags them with an annotation at e.g. https://github.com/spack/spack/actions/runs/9159378954.

We probably also should ignore `.dat` files (one is flagged as well) but I've filed a feature request to remove that particular one.

Reply:
I don't think there is a way to entirely skip the files which do not decode as utf-8 or to specify which encoding to use per file . ATM it is hardcoded to try        `for encoding in ("utf-8", "iso-8859-1"):` . And thus it does catch/fix typos already.  We can add pragma to ignore specific lines, but that wouldn't ignore entire file reading first.

Filed
- https://github.com/codespell-project/codespell/issues/3427

and feel welcome to comment on if you see some other desired way.

re .dat -- sure we could ignore those if you like. pushed